src/Microsoft.IdentityModel.Tokens/CryptoProviderFactory.cs

```diff
@@ -256,6 +279,37 @@ public virtual AuthenticatedEncryptionProvider CreateAuthenticatedEncryptionProv
         }
 
         /// <summary>
+        /// Creates an instance of <see cref="KeyWrapProvider"/> for a specific &lt;SecurityKey, Algorithm>.
+        /// </summary>
+        /// <param name="key">the <see cref="SecurityKey"/> to use.</param>
+        /// <param name="algorithm">the algorithm to use.</param>
+        /// <returns>an instance of <see cref="KeyWrapProvider"/></returns>
+        /// <exception cref="ArgumentNullException">'key' is null.</exception>
+        /// <exception cref="ArgumentNullException">'algorithm' is null or empty.</exception>
+        /// <exception cref="ArgumentException">If <see cref="SecurityKey"/> and algorithm pair are not supported.</exception>
+        /// <exception cref="ArgumentException">'key' is not a <see cref="SymmetricSecurityKey"/>.</exception>
+        public virtual KeyWrapProvider CreateKeyWrapProvider(SecurityKey key, string algorithm)
+        {
+            if (key == null)
+                throw LogHelper.LogArgumentNullException(nameof(key));
+
+            if (string.IsNullOrEmpty(algorithm))
+                throw LogHelper.LogArgumentNullException(nameof(algorithm));
+
+            if (CustomCryptoProvider != null && CustomCryptoProvider.IsSupportedAlgorithm(algorithm, key))
```

Comment:
We allow user to provide their own CryptoProvider to create their own KeyWrapProvider. We follow the same pattern in all the similar methods like CreateAuthenticatedEncryptionProvider.